drivers/hmis_csv_importer/app/models/hmis_csv_importer/importer/importer.rb

```diff
@@ -101,22 +108,224 @@ def resume!
+
+      any_record_count_thresholds_met? && @data_source.ever_pause_imports_with_record_changes?
+    end
 
-      db_errors = HmisCsvImporter::Importer::ImportError.where(
+    ##
+    # Determines if any file in the import process has exceeded its allowed error threshold.
+    #
+    # This method iterates over all import files and checks whether the number of errors
+    # in each file surpasses the defined threshold. If any file's error count exceeds the
+    # threshold set in the data source, the method returns `true`.
+    #
+    # @return [Boolean] `true` if any file has met or exceeded its error threshold, otherwise `false`.
+    #
+    memoize private def any_error_thresholds_met?
+      return false unless @data_source.ever_notify_for_imports?
+
+      # counts of expected rows in each file
+      totals_by_filename = @loader_log.summary.map { |filename, data| [filename, data['total_lines'].to_i] }.to_h
+      totals_by_filename.each do |filename, total_count|
+        met = @data_source.error_count_threshold_reached?(total_count, error_counts[filename])
+
+        return true if met
+      end
+      # No threshold met
+      false
+    end
+
+    ##
+    # Determines if any file in the import process has exceeded its allowed record count change threshold.
+    #
+    # This method evaluates the total number of records and absolute number of changes (additions minus removals)
+    # for each file. If the number of changes exceeds the configured threshold for any file, the method
+    # returns `true`.
+    #
+    # @return [Boolean] `true` if any file has met or exceeded its record count change threshold, otherwise `false`.
+    #
+    memoize private def any_record_count_thresholds_met?
+      return false unless @data_source.ever_notify_for_imports?
+
+      change_counts.each_value do |data|
+        total = data[:total_count]
+        changes = data[:change_count]
+        met = @data_source.record_count_threshold_reached?(total, changes)
+
+        return true if met
+      end
+      # No threshold met
+      false
+    end
+
+    ##
+    # Collects and aggregates import errors from the loader and import processes.
+    #
+    # This method retrieves error counts from different parts of the import process:
+    # - Errors logged in the loader summary.
+    # - Errors recorded in the {HmisCsvImporter::Importer::ImportError} model.
+    # - Validation errors recorded in the {HmisCsvImporter::HmisCsvValidation::Base} model.
+    #
+    # It maps error counts to their respective filenames and ensures that the counts
+    # are aggregated correctly without raising additional errors if files are missing.
+    #
+    # @return [Hash{String => Integer}] A hash where keys are filenames and values are the total error counts for each file.
+    #
+    memoize private def error_counts
+      file_lookup = self.class.importable_files_map.invert
+
+      summary = @loader_log.summary
```

Comment:
for my own understanding, I think this is right?
```suggestion
      # serialized array persisted on the log model
      summary = @loader_log.summary
```